src/neuroglancer/ui/annotations.ts

```diff
@@ -1333,7 +1334,7 @@ export function UserLayerWithAnnotationsMixin<TBase extends {new (...args: any[]
         boolean {
       if (state.annotationId === undefined) return false;
       const annotationLayer = this.annotationStates.states.find(
-          x => x.sourceIndex === state.annotationSourceIndex &&
+          x => x.subsourceIndex === state.annotationSubsourceIndex &&
```

Comment:
This temp hack was made because there was an issue with the selection side panel displaying "Annotation not found" on the second annotationState in the same AnnotationLayerView. I don't know the proper way to handle this.

Reply:
It would be simpler if the annotations were stored in a single annotation source, since the existing annotation code assumes a single annotation source per subsource.  However, I guess the reason for two annotation sources is so that they can be displayed in separate colors, and also shown in separate lists.  I think there are two possible solutions:

1. Use a single annotation source for both "red" and "blue" points.  The colors could be handled via an annotation property.  Some change would be needed to filter them in the list, though.
2. Add an additional optional string identifier in order to identify multiple annotation sources within the same subsource --- basically the same as you have here, except using `string | undefined` as the identifier type in place of `number`.

I think option 2 is probably the best way to go.

Reply:
For option 2, any idea what I should call it? I was going to name it annotationSubsourceId but in annotations.ts it has.  `state.annotationSubsource = annotationLayer.subsourceId` I could rename the existing AnnotationLayerState.subsourceId to just subsource.

What does AnnotationLayerState currently use subsourceId and subsourceIndex for?